protocol/encryption/encryptor.go

```diff
@@ -592,6 +596,132 @@ func (s *encryptor) EncryptPayload(theirIdentityKey *ecdsa.PublicKey, myIdentity
 	return response, targetedInstallations, nil
 }
 
+func (s *encryptor) getNextKeyID(communityId string) (uint32, error) {
+	latestKeyId, err := s.persistence.GetCurrentKeyForCommunity(communityId)
+	if err != nil {
+		return 0, err
+	}
+	currentTime := (uint32)(time.Now().UnixNano() / int64(time.Millisecond))
+	keyIdBump := (uint32)(10)
+	if latestKeyId < currentTime {
+		return currentTime + keyIdBump, nil
+	} else {
+		return latestKeyId + 1, nil
+	}
+}
+
+// Generates and stores a community key
+func (s *encryptor) GenerateCommunityKey(communityId []byte) (uint32, error) {
```

Comment:
I would not mention `Community` in general, we can use `GenerateHashRatchetKey` for example